ddtrace/contrib/graphql/patch.py

```diff
@@ -0,0 +1,84 @@
+from typing import Union
+
+import graphql
+from graphql.language import Source
+
+from ddtrace import Span
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.constants import SPAN_MEASURED_KEY
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils.wrappers import unwrap as _u
+from ddtrace.pin import Pin
+from ddtrace.vendor.wrapt import wrap_function_wrapper as _w
+
+from .. import trace_utils
+from ...ext import SpanTypes
+
+
+config._add("graphql", dict(_default_service="graphql"))
+
+graphql_version_str = getattr(graphql, "version", "0.0.0")
+graphql_version = tuple([int(i) for i in graphql_version_str.split(".")])
+
+
+def patch():
+    if getattr(graphql, "_datadog_patch", False):
+        return
+    elif graphql_version < (1, 1):
+        return
+
+    setattr(graphql, "_datadog_patch", True)
+    pin = Pin()
+    _w(graphql, "graphql", _traced_graphql)
+    _w(graphql, "graphql_sync", _traced_graphql_sync)
+    pin.onto(graphql)
+
+
+def unpatch():
+    if not getattr(graphql, "_datadog_patch", False):
+        return
+    elif graphql_version < (1, 1):
+        return
+
+    setattr(graphql, "_datadog_patch", False)
+    _u(graphql, "graphql")
+    _u(graphql, "graphql_sync")
+
+
+async def _traced_graphql(func, instance, args, kwargs):
+    return await _traced_graphql_sync(func, instance, args, kwargs)
```

Comment:
does this work? `_traced_graphql_sync` isn't an async function... does it return a task?

Reply:
I believe it should work. I can refactor this if it's hard to follow. I added a test case which uses an async resolver.

`graphql.graphql(schema, source, ..., awaitable=False)` can return either an Awaitable or an ExecutionResult object : https://github.com/graphql-python/graphql-core/blob/main/src/graphql/graphql.py#L92-L95.  While `graphql.graphql_sync` always returns an ExecutionResult object. 

Both methods return the result of `graphql_impl()` which is a synchronous function: https://github.com/graphql-python/graphql-core/blob/main/src/graphql/graphql.py#L152. I could just instrument this method instead but it's not part of the public API so it might change/break in a minor version.

Reply:
apologies for assuming it didn't work, it did look a little off to me, maybe it is just the naming (async function awaits a sync function) that threw me off